### PR TITLE
test: add e2e tests for manifest fetch-config under oci-layout

### LINKED ISSUE #864: add OCI image layout e2e test specs for `oras manifest fetch-config`

### PR DESCRIPTION
**Pull request by wangxiaoxuan273**
Resolves #864

### CODE REVIEW

File: test/e2e/suite/command/manifest.go [hunk 1]
```diff
@@ -367,3 +367,47 @@ var _ = Describe("Common registry users:", func() {
...
+var _ = Describe("OCI image layout users:", func() {
+	prepare := func(from string, tag string) string {
+		tmpRoot := GinkgoT().TempDir()
+		cpPath := tmpRoot
+		if tag != "" {
+			cpPath = fmt.Sprintf("%s:%s", tmpRoot, tag)
+		}
+		ORAS("cp", from, Flags.ToLayout, cpPath).WithDescription("prepare image from registry to OCI layout").Exec()
+		return tmpRoot
+	}
```
Comment (qweeah) [inline test/e2e/suite/command/manifest.go]: refactor to `prepare := func(tag string) string`

**Review by qweeah — CHANGES_REQUESTED** [review]
Can you help also add spec covering below bad paths:
- should fail fetching manifest config from registry without any tag or digest provided
- should fail fetching manifest config from layout without any tag or digest provided
- should fail fetching manifest from registry without any tag or digest provided
- should fail fetching manifest from layout without any tag or digest provided

File: test/e2e/suite/command/manifest.go (hunk 1 above)
Comment (wangxiaoxuan273) [reply test/e2e/suite/command/manifest.go]: done and added cases for bad paths.

**Review by qweeah — APPROVED** [review]
LGTM

**Pull request merged**